### DISCUSSION TITLE
Configure multiple push url for one origin

### DISCUSSION THREAD

**Discussion by aminnairi** [NeogitOrg/neogit#1105] [Q&A]
Hi, and thank you for this awesome plugin!

Using the Git CLI, I can configure multiple push url for a single origin. This is quite interesting, especially when using multiple cloud providers in order to backup our work in case of a catastrophic failure.

In order to do that, we can use the following command.

```bash
git remote set-url --push --add origin git@github.com:aminnairi/repo.git
git remote set-url --push --add origin git@gitlab.com:aminnairi/repo.git
```

That way, in a single push, I can have my work saved to many cloud provider.

I notice it is possible to add a `pushUrl` using `neogit`, but it looks like only one pushUrl is allowed, whereas using `git`, we can set up mulitple push URLs.

![image](https://github.com/NeogitOrg/neogit/assets/18418459/f3d42bf3-50a0-4c2a-8534-27b485525e85)

In this case (after using the `git` CLI to add multiple push URLs), Neogit isn't capable of displaying them all here.

Maybe there is a special syntax that allows delimiting multiple URLs?

If so, I would like to know!

If not, I would like to know if this is a feature that could be added to neogit.

Thank you!

P.S. : I would also like to know the difference between `push` and `pushUrl` here.

**Comment by CKolkey** [comment]
Hey! Thanks for pointing this out - I had no idea. Checking how Magit handles it, 
<img width="503" alt="Screenshot 2024-01-04 at 21 58 09" src="https://github.com/NeogitOrg/neogit/assets/7228095/24ca065f-8b93-4234-95eb-b167997e6829">

It's pretty clearly a thing we should do too. I'll have to re-do a bit of how the config stuff works, since I made the assumption that keys were unique, but... I'll sort that out.

**Reply by CKolkey** [reply]
Here's the details on the differences between `push` and `pushUrl`
https://magit.vc/manual/magit/Remote-Git-Variables.html

:)

**Reply by aminnairi** [reply]
I was today's year old when I figured we could have multiple push URLs for a single origin :rofl: 

I'll let you sort that out since I'm not (yet) capable of contributing to Neogit. Hope that you'll figure this out and eager to have this feature soon now that I know we can do that with Git!

Thank you for your prompt reply, and thank you(among others) for making this plugin available to Neovim, a clear game changer :fire: 

And thank you for pointing me to the right documentation :pray:
